crates/polars-io/src/csv/write_impl.rs

```diff
@@ -186,18 +186,21 @@ unsafe fn write_anyvalue(
                                 },
                                 _ => ndt.format(datetime_format),
                             };
-                            return write!(f, "{formatted}").map_err(|_|{
-
+                            write!(f, "{formatted}").map_err(|_|{
                                 let datetime_format = unsafe { *datetime_formats.get_unchecked(i) };
                                 let type_name = if tz.is_some() {
                                     "DateTime"
                                 } else {
                                     "NaiveDateTime"
                                 };
                                 polars_err!(
-                ComputeError: "cannot format {} with format '{}'", type_name, datetime_format,
-            )
-                            });
+                                    ComputeError: "cannot format {} with format '{}'", type_name, datetime_format,
+                                )
+                            })?;
+                            if end_with_quote {
+                                write!(f, "{quote}")?
+                            }
+                            return Ok(())
```

Comment:
I think we can just omit this? Just removing the return should be enough, right? :thinking:

Reply:
There may be a more streamlined way to do it, but you can't just omit or it won't satisfy the `match` arms (I started off that way but it started becoming somewhat contorted; was probably written using `return` originally for much the same reason).

Reply:
I'll take a look. I don't like the code duplication introduced here.

Reply:
Ahhh... switching to `polars_bail` was staring me right in the face, hah... good call :)